app_flutter/lib/service/appengine_cocoon.dart

```diff
@@ -46,6 +46,48 @@ class AppEngineCocoonService implements CocoonService {
     return _commitStatusesFromJson(jsonResponse['Statuses']);
   }
 
+  @override
+  Future<bool> fetchTreeBuildStatus() async {
+    /// This endpoint returns JSON {AnticipatedBuildStatus: [BuildStatus]}
+    http.Response response =
+        await _client.get('$_baseApiUrl/public/build-status');
+
+    if (response.statusCode != HttpStatus.ok) {
+      throw HttpException(
+          '$_baseApiUrl/public/build-status returned ${response.statusCode}');
+    }
+
+    Map<String, Object> jsonResponse = jsonDecode(response.body);
+
+    if (!_isBuildStatusResponseValid(jsonResponse)) {
+      throw HttpException(
+          '$_baseApiUrl/public/build-status had a malformed response');
+    }
+
+    return jsonResponse['AnticipatedBuildStatus'] == "Succeeded";
+  }
+
+  /// Check if [Map<String,Object>] follows the format for build-status.
+  ///
+  /// Template of the response it should receive:
+  /// ```json
+  /// {
+  ///   "AnticipatedBuildStatus": "Succeeded"|"Failed"
+  /// }
+  /// ```
+  bool _isBuildStatusResponseValid(Map<String, Object> response) {
+    if (!response.containsKey('AnticipatedBuildStatus')) {
+      return false;
+    }
+
+    String treeBuildStatus = response['AnticipatedBuildStatus'];
+    if (treeBuildStatus != 'Failed' && treeBuildStatus != 'Succeeded') {
+      return false;
+    }
```

Comment:
Nit: it's more readable to write this in a positive way:
```
if (treeBuildStatus == "Succeeded" || treeBuildStatus == "Failed") {
  return true;
}

return false;
```

"In general checking the presence or absence of a positive is simpler for readers to understand than checking the presence or absence of a negative."

Reply:
For future reference: Testing on the Toilet #530: Improve Readability with Positive Booleans

The reason I chose this way was to have cases that returned false next to each other, instead of case 1 returns false, case 2 returns true, final case returns false. I wanted to try and reduce the cognitive switching on the return value.